x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_preview_rule.ts

```diff
@@ -137,5 +142,5 @@ export const usePreviewRule = ({
     };
   }, [rule, addError, invocationCount, from, interval, timeframeEnd]);
 
-  return { isLoading, showInvocationCountWarning, response, rule, setRule };
+  return { isLoading, isInvalidInterval, showInvocationCountWarning, response, rule, setRule };
```

Comment:
Nit: This is not related to this PR, but as an offtopic suggestion, I'd like to suggest refactoring this hook by extracting some blocks into smaller functions and leveraging `react-query` for calling the preview endpoint. This could simplify this piece a lot:

```ts
  useEffect(() => {
    let isSubscribed = true;
    const abortCtrl = new AbortController();
    setResponse(emptyPreviewRule);
    const createPreviewId = async () => {
      if (rule != null) {
        try {
          setIsLoading(true);
          const previewRuleResponse = await previewRule({
            rule: {
              ...transformOutput({
                ...rule,
                interval,
                from,
              }),
              invocationCount,
              timeframeEnd,
            },
            signal: abortCtrl.signal,
          });
          if (isSubscribed) {
            setResponse(previewRuleResponse);
          }
        } catch (error) {
          if (isSubscribed) {
            addError(error, { title: i18n.RULE_ADD_FAILURE });
          }
        }
        if (isSubscribed) {
          setIsLoading(false);
        }
      }
    };

    createPreviewId();
    return () => {
      isSubscribed = false;
      abortCtrl.abort();
    };
  }, [rule, addError, invocationCount, from, interval, timeframeEnd]);
```

We've been using `react-query` in the Rules area for a while and can recommend it. Feel free to ping me or @xcrzx on that.

Reply:
I removed related changes here, so I would prefer to make the changes that you mentioned in a separate PR in 8.5 to avoid introducing potential bugs in 8.4 release.